dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/InjectionUtils.java

```diff
@@ -971,13 +971,32 @@ private static Object mergeCollectionsOrArrays(Object first, Object second, Type
     }
 
     //CHECKSTYLE:OFF
+    @FFDCIgnore(value = { IndexOutOfBoundsException.class }) // Liberty Change
     private static Object injectIntoCollectionOrArray(Class<?> rawType,
                                                       Type genericType,
                                                       Annotation[] paramAnns,
                                                       MultivaluedMap<String, String> values,
                                                       boolean isbean, boolean decoded,
                                                       ParameterType pathParam, Message message) {
         //CHECKSTYLE:ON
+        // Liberty change start
+        ParamConverter<?> pm = null;
+        if (message != null) {
+            ServerProviderFactory pf = ServerProviderFactory.getInstance(message);
+            pm = pf.createParameterHandler(rawType, genericType, paramAnns, message);
+            if (pm != null) {
+                for (Map.Entry<String, List<String>> entry : values.entrySet()) {
+                    // always only ever 1?
```

Comment:
I would suggest checking the size of "values" and putting a debug trace in here indicating that there are more entries than we expect and listing them